src/apps/src/eryph-packer/Program.cs

```diff
@@ -257,50 +284,56 @@
                 var validationResult = CatletConfigValidations.ValidateCatletConfig(catletConfig);
-                validationResult.ToEither()
+                _ = validationResult.ToEither()
                     .MapLeft(issues => Error.New("The catlet configuration is invalid.",
                         Error.Many(issues.Map(i => i.ToError()))))
                     .IfLeft(e => e.Throw());
                 var configJson = ConfigModelJsonSerializer.Serialize(catletConfig);
                 await File.WriteAllTextAsync(Path.Combine(packFolder, "catlet.json"), configJson);
                 packableFiles.Add(new PackableFile(Path.Combine(packFolder, "catlet.json"),
-                    "catlet.json", GeneType.Catlet, "catlet", false));
+                    "catlet.json", GeneType.Catlet,
+                    Architectures.Any, // catlet is architecture independent
+                    "catlet", false, catletContent));
                 parent = catletConfig.Parent;
             }
 
             AnsiConsole.MarkupLine("Catlet [green]prepared[/]");
 
             // pack fodder
             AnsiConsole.MarkupLine("Preparing fodder...");
-            
+
             var fodderDir = new DirectoryInfo(Path.Combine(absoluteGenesetPath, "fodder"));
             if (fodderDir.Exists)
             {
-                foreach (var fodderFile in fodderDir.GetFiles("*.*").Where(x =>
-                         {
-                             var extension = Path.GetExtension(x.Name).ToLowerInvariant();
-                             return extension is ".yaml" or ".yml";
-                         }))
+                await AddFodderFromDirectory(fodderDir, Architectures.Any);
+                foreach (var hypervisorDir in fodderDir.GetDirectories())
                 {
-                    var fodderContent = File.ReadAllText(fodderFile.FullName);
-                    var fodderConfig = DeserializeFodderConfigString(fodderContent);
-                    var validationResult = FodderGeneConfigValidations.ValidateFodderGeneConfig(fodderConfig);
-                    validationResult.ToEither()
-                        .MapLeft(issues => Error.New($"The fodder configuration '{fodderFile.Name}' is invalid.",
-                            Error.Many(issues.Map(i => i.ToError()))))
-                        .IfLeft(e => e.Throw());
-                    var fodderJson = ConfigModelJsonSerializer.Serialize(fodderConfig);
-                    var fodderPackFolder = Path.Combine(packFolder, "fodder");
-                    if (!Directory.Exists(fodderPackFolder))
-                        Directory.CreateDirectory(fodderPackFolder);
+                    if (string.Equals(hypervisorDir.Name, Hypervisors.HyperV, StringComparison.OrdinalIgnoreCase))
+                    {
+                        await AddFodderFromDirectory(fodderDir, Architectures.HyperVAny);
+
+                        foreach (var archDir  in fodderDir.GetDirectories())
```

Comment:
Should be `... hypervisorDir.GetDirectories` (assuming that we use a nested directory structure).

Reply:
jupp, changed to processortype, see below